<launch>
    <!-- 
        Launches Quadcopter Gazebo simulation with geometric controller anf circular trajectory publisher
     -->

    <arg name="command_input" default="2" />
    <arg name="gazebo_simulation" default="true" />
    <arg name="visualization" default="true"/>

    <node pkg="geometric_controller" type="geometric_controller_node" name="geometric_controller" output="screen">
        <!-- <param name="mav_name" type="string" value="$(arg mav_name)" /> -->
        <remap from="command/bodyrate_command" to="/mavros/setpoint_raw/attitude"/>
        <param name="ctrl_mode" value="$(arg command_input)" />
        <param name="enable_sim" value="$(arg gazebo_simulation)" />
        <param name="enable_gazebo_state" value="true"/>
        <param name="max_acc" value="10.0" />
        <!--Params specific for Iris copter-->
        <param name="attctrl_constant" value="0.3"/>
        <param name="normalizedthrust_constant" value="0.06"/>
        <param name="normalizedthrust_offset" value="0.1"/>
        <param name="Kp_x" value="10.0" />
        <param name="Kp_y" value="10.0" />
        <param name="Kp_z" value="20.0" />
        <param name="Kv_x" value="5.0"/>
        <param name="Kv_y" value="5.0"/>
        <param name="Kv_z" value="10.0"/>
    </node>

    <node pkg="trajectory_publisher" type="trajectory_publisher" name="trajectory_publisher" output="screen">
        <param name="trajectory_type" value="1" />
        <param name="shape_omega" value="1.2" />
        <param name="initpos_z" value="2.0" />
        <param name="reference_type" value="2" />
    </node>

    <!-- Launch rqt_reconfigure -->
    <!-- <node pkg="rqt_reconfigure" type="rqt_reconfigure" output="screen" name="rqt_reconfigure" /> -->

    <group if="$(arg visualization)">
        <node type="rviz" name="rviz" pkg="rviz" args="-d $(find geometric_controller)/launch/config_file.rviz" />
    </group>
</launch>